src/python/deepgnn/graph_engine/snark/client.py

```diff
@@ -224,6 +256,10 @@ def __init__(
         )
         self._describe_clib_functions()
 
+        self.dist_mode = False # used for DistGraph only
+        self.retries = None # used for DistGraph only
```

Comment:
Can you use a retry library instead? I suggest tenacity: https://github.com/jd/tenacity
You can have a custom retryer field that can wrap existing functions and implements jitter and exponential backoff. It will avoid code duplication then. Something like this:
```python
class MemoryGraph:
    def __init__ (self, ...):
        self._retryer = Retrying(stop=stop_after_attempt(max_attempts=0), reraise=True)
        ...
        def node_features(...):
            ....
            self._retryer(self.lib.GetNodeFeature, ...)
            
class DistributedGraph:
    def __init__ (self, ...):
        retry_options = _parse_grpc_options(...)
        self._retryer = Retrying(stop=stop_after_attempt(max_attempts=retry_options['retries']), reraise=True)
```

Reply:
So we have retry for MemoryGraph (non-distributed) as well?

Reply:
Well, it will be a no-op retry.